integration/users/connection_test.go

```diff
@@ -412,3 +414,49 @@ func TestAuthenticationEnableNewAuthPLAIN(t *testing.T) {
 		})
 	}
 }
+
+func TestAuthenticationSASLContinueFailure(t *testing.T) {
+	t.Parallel()
+
+	s := setup.SetupWithOpts(t, nil)
+	collection := s.Collection
+	db := collection.Database()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	require.NoError(t, db.RunCommand(ctx, bson.D{
+		{"createUser", "pencil"},
+		{"roles", bson.A{}},
+		{"pwd", "password"},
+	}).Err())
+
+	credential := options.Credential{
+		AuthMechanism: "SCRAM-SHA-1",
+		AuthSource:    db.Name(),
+		Username:      "pencil",
+		Password:      "password",
+	}
+
+	opts := options.Client().ApplyURI(s.MongoDBURI).SetAuth(credential)
+
+	client, err := mongo.Connect(ctx, opts)
+	require.NoError(t, err, "cannot connect to MongoDB")
+
+	dbNew := client.Database(db.Name())
+
+	err = dbNew.Client().Ping(context.Background(), nil)
+	require.NoError(t, err)
+
+	err = dbNew.RunCommand(ctx, bson.D{
+		{"saslContinue", int32(1)},
+		{"conversationId", int32(1)},
+		{"payload", []byte("Yz1iaXdzLHI9UnVBWkJSQVpnL1JlZ1FSamF2d2ZqNmhVRDBSUXRJL2VaUTQvemdBUU9NZk83dDdDUGRsR3hnQW9JQkpEODYrMixwPU52VTFQWW1GcldUT1N2ZVBFb05SdDVuMkplNHc2aVBMS2phL2phSC9XZmc9")},
+	}).Err()
+
+	integration.AssertEqualCommandError(t, mongo.CommandError{
+		Name:    "AuthenticationFailed",
+		Code:    18,
+		Message: "Authentication failed.",
+	}, err)
+}
```

Comment:
So, this is what calling the saslContinue method explicitly like that would return, and even though the test is passing on FerretDB with PostgreSQL/SQLite, the failure comes from a different code path, so this is a false positive. We need to call saslContinue on the exact same moment as it happens when using mongosh, and this isn't the case here: that's why I said we can't get away with just calling the method.
/cc @AlekSi

Reply:
So the error comes from different code path because, `saslStart` passed successfully so conv is not nil. When `saslContinue` is called, it fails because payload is wrong.

I tried another way using invalid user credentials so that `saslStart` fails (conv is nil), then call `saslContinue` but the driver does not allow calling `saslContinue` before successful `saslStart`. So it fails with `saslStart` error and doesn't test what we want.